### DISCUSSION TITLE
Support previewing variables

### DISCUSSION THREAD

**Discussion by iainlane** [grafana/grafana#58185] [Alerting]
When using variables in alert annotations, if you get the variable name a bit wrong, the verbatim string is substituted instead.

  1. I can't find anywhere in the Grafana UI that this happened, it would be good to show a warning as a diagnostic that the templating failed to render and we fell back to sending the raw value.
  2. It can be a bit cumbersome to iterate in this way: make change, make alert fire, test in your receiver (email, Slack, wherever) that it looks right. Suggestion: how about adding a preview feature so that you can see what your alert would look like, to check if you got the templating correct or not.

I guess the preview would only work if the alert would fire or be pending right away (otherwise we wouldn't know what values would be available to substitute), so it might not be very simple to come up with an easy UX for this.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
